Index: src/mac/carbon/clipbrd.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/clipbrd.cpp,v
retrieving revision 1.35
diff -u -4 -r1.35 clipbrd.cpp
--- src/mac/carbon/clipbrd.cpp	2005/02/18 04:43:45	1.35
+++ src/mac/carbon/clipbrd.cpp	2005/04/18 23:06:12
@@ -250,8 +250,9 @@
             {
                case wxDF_TEXT:
                case wxDF_OEMTEXT:
                     mactype = kScrapFlavorTypeText ;
+                    sz -= 1;
                break ;
     #if wxUSE_UNICODE
                case wxDF_UNICODETEXT :
                     mactype = kScrapFlavorTypeUnicode ;
@@ -259,8 +260,9 @@
     #endif
     #if wxUSE_DRAG_AND_DROP
             case wxDF_METAFILE:
                     mactype = kScrapFlavorTypePicture ;
+                    sz -= 2;
                break ;
     #endif
                case wxDF_BITMAP:
                case wxDF_DIB:
